plugins/woocommerce/includes/wc-notice-functions.php

```diff
@@ -137,7 +137,14 @@ function wc_print_notices( $return = false ) {
 		return;
 	}
 
-	$all_notices  = WC()->session->get( 'wc_notices', array() );
+	$session = WC()->session;
+
+	// If the session handler has not initialized, there will be no notices for us to read.
+	if ( null === $session ) {
+		return null;
```

Comment:
I think the `@return` clause in the doc block for this is wrong. It seems like it should be `string|void`, since that's what it does when you echo the content instead of returning it. As such, this should probably just be:

```suggestion
		return;
```

Reply:
Sharp eyes, great catch! Updated in https://github.com/woocommerce/woocommerce/pull/40648/commits/4a1b225571a42b500f26f69de6d00bd412f52bba.